src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -515,6 +518,11 @@ private function parseDefinition($id, $service, $file, array $defaults)
             }
         }
 
+        $autoConfigureInstanceof = isset($service['auto_configure_instanceof']) ? $service['auto_configure_instanceof'] : (isset($defaults['auto_configure_instanceof']) ? $defaults['auto_configure_instanceof'] : null);
```

Comment:
Nested ternaries are hard to read :) Why not using `if` blocks? In Symfony 4 we'll be able to replace such constructs with `??`.